internal/clientconn/conninfo/conn_info.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package conninfo provides a ConnInfo struct that is used to handle connection-specificinfo
+// and can be shared through context.
+package conninfo
+
+import (
+	"context"
+	"net"
+)
+
+// contextKey is a special type to represent context.WithValue keys a bit more safely.
+type contextKey string
```

Comment:
```suggestion
type contextKey struct{}
```

That makes it even safer